packages/wonder-blocks-data/src/hooks/use-data-internal.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+import {Server} from "@khanacademy/wonder-blocks-core";
+import {useState, useEffect, useContext, useRef} from "react";
+import {RequestFulfillment} from "../util/request-fulfillment.js";
+import {TrackerContext} from "../util/request-tracking.js";
+import {resultFromCacheEntry} from "../util/result-from-cache-entry.js";
+import {ResponseCache} from "../util/response-cache.js";
+
+import type {
+    Result,
+    IRequestHandler,
+    ValidData,
+    CacheEntry,
+} from "../util/types.js";
+
+export const useDataInternal = <TOptions, TData: ValidData>(
+    handler: IRequestHandler<TOptions, TData>,
+    options: TOptions,
+): Result<TData> => {
+    // If we're server-side or hydrating, we'll have a cached entry to use.
+    // So we get that and use it to initialize our state.
+    // This works in both hydration and SSR because the very first call to
+    // this will have cached data in those cases as it will be present on the
+    // initial render - and subsequent renders on the client it will be null.
```

Comment:
All the ones that were used, which should be all of them. If it isn't, then hydration is broken because we requested stuff server-side for things that don't get rendered client-side.